extensions/eclipse-che-theia-plugin-ext/src/browser/che-telemetry-main.ts

```diff
@@ -46,6 +49,18 @@ export class CheTelemetryMainImpl implements CheTelemetryMain {
             }
         }
 
-        return this.cheApiService.submitTelemetryEvent(id, ownerId, ip, agent, resolution, properties);
+        return this.cheApiService.submitTelemetryEvent(id, ownerId, this.ip, agent, resolution, properties);
+    }
+
+    async $getClientAddressInfo(): Promise<ClientAddressInfo> {
+       return this.getClientAddressInfo();
+    }
+
+    async getClientAddressInfo(): Promise<ClientAddressInfo> {
+        const response = await axios.default.get('/che/client-ip');
+        if (response.status === 200) {
+            return response.data;
+        }
+        return {ip: 'not defined'};
```

Comment:
it should be `undefined` type (rather than 'not defined' string)
Also might be interesting to log the error if we don't get correct status/error code (or include status in clientAddressInfo)